### PR TITLE
Add clojure test generator and clock exercise

### PR DESCRIPTION
**Pull request by haus**
This commit adds a generator namespace, adds a stencil dependency, and
adds x-common to the .gitignore. When invoked, this will clone the
x-common repo if not present, then load and munge any test-data for the
passed exercise, and then render clojure tests using a mustache template
for the exercise.

### CODE REVIEW

**Comment by haus** [comment]
Happy to break this into 2 PRs if that makes more sense. I put them together to show the how the test generation would work.

File: _src/generator.clj [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+(defn file-exists?
+  "Helper function to determine if a given file exists."
+  [path]
+  (.exists (File. path)))
+
+(defn clone-test-data
+  "Clone the x-common repo from github"
+  []
+  (sh "git" "clone" "git@github.com:exercism/x-common"))
```
Comment (haus) [inline _src/generator.clj]: It looks like the ruby test generator doesn't actually clone the repo, but just warns and exits if the repo doesn't exist (https://github.com/exercism/xruby/blob/master/lib/generator.rb#L46). Would that be preferable here?

File: project.clj [hunk 2]
```diff
@@ -2,5 +2,8 @@
   :description  "Exercism Exercises in Clojure"
   :url          "https://github.com/exercism/xclojure"
   :test-paths   ["_test"]
+  :source-paths ["_src"]
```
Comment (haus) [inline project.clj]: I used `_src` here because `_test` had been used for tests previously. Happy to put it somewhere else as well.

**Comment by haus** [comment]
Looks like travis may be sad because i added the _src directory.

**Comment by yurrriq** [comment]
Looks like you need to add it to `config.json` maybe.

File: _src/generator.clj [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+       (format "Could not find test data for %s (looking in %s)"
+               exercise-name test-data-filename)))
+    (json/parse-stream (reader test-data-filename))))
+
+(defn munge-test-data
+  "Loads the generator namespace for the exercise and calls the munge-data function on the given test-data."
+  [exercise-name test-data]
+  (let [exercise-ns (symbol (str exercise-name "-generator"))]
+    (require [exercise-ns])
+    (if-let [munge-data-fn (ns-resolve exercise-ns (symbol "munge-data"))]
```
Comment (haus) [inline _src/generator.clj]: In the case of the clock exercise munging the canonical data wasn't necessary, but it seemed like a good escape hatch to leave in for other exercises, as mustache is purely data driven templating.

**Comment by yurrriq** [comment]
I think this should be good to merge now. Go ahead if you agree.

**Comment by haus** [comment]
Right on.

**Pull request merged**